File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/Message.m
//
//  Message.m
//  OpenVideoCall
//
//  Created by CavanSu on 24/11/2017.
//  Copyright © 2017 Agora. All rights reserved.
//

#import "Message.h"

@implementation Message
+ (instancetype)messageWithText:(NSString *)text type:(MessageType)type {
    Message *message = [[Message alloc] init];
    message.text = text;
    message.type = type;
    return message;
}
@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/Message.swift
//
//  Message.swift
//  OpenVideoCall
//
//  Created by GongYuhua on 16/8/19.
//  Copyright © 2016年 Agora. All rights reserved.
//

import UIKit

enum MessageType {
    case info, alert
    
    func color() -> UIColor {
        switch self {
        case .info: return UIColor(hex: 0x444444, alpha: 0.6)
        case .alert: return UIColor(hex: 0xff3c32, alpha: 0.6)
        }
    }
}

struct Message {
    var text: String
    var type: MessageType = .info
    
    init(text: String, type: MessageType) {
        self.text = text
        self.type = type
    }
}
    
